### PR TITLE
Improve unit tests

### PR DESCRIPTION
**Pull request by johnBartos**
Downstream PR of https://github.com/video-dev/hls.js/pull/1923

### This PR will...
- Replace Node `assert` with Chai `expect`. `expect` must now be used in place of `assert`
- Refactor tests to use [Chai](https://www.chaijs.com/api/bdd/) and [Sinon-Chai](https://www.chaijs.com/plugins/sinon-chai/) BDD assertions
- Remove`should`, `nyc`, `arraybuffer-equal`, and `deep-strict-equal` from tests & package.json
- Avoid preprocessing `node_modules` with webpack so that Karma can successfully load plugins
- Bundle all tests into a single file via `require.context` in `index.js`
- Bumps dependencies to the latest version

### Why is this Pull Request needed?

1) Speed. Tests complete 3x faster with these changes. Bundling tests into a single file is twice as fast as not:

Before:
Without any changes

Refactored but unbundled:
Refactor but unbundled

Refactored and bundled:
After refactoring, with bundling

2) Developer ergonomics. Chai allows developers to write more complex assertions with less code. BDD is also easier to understand for someone who isn't familiar with our codebase. This should allow developers to more easily write unit tests which should increase our test coverage in the long run.

### Are there any points in the code the reviewer needs to double check?
I still need to make sure sourcemaps are working fine

### Resolves issues:
Just something I've been wanting to do.

### Checklist

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] API or design changes are documented in API.md

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -12,20 +12,21 @@ module.exports = function(config) {
 
 		// frameworks to use
 		// available frameworks: https://npmjs.org/browse/keyword/karma-adapter
-		frameworks: ['mocha', 'sinon'],
+		frameworks: ['mocha', 'sinon-chai'],
 
 		// list of files / patterns to load in the browser
 		files: [
-      'tests/unit/**/*.js'
```
Comment (robwalch) [inline karma.conf.js]: What's better about adding this index file? Are there advantages and/of side-effects of how webpack bundles the test modules?

**Comment by johnBartos** [comment]
Blocked by https://github.com/jwplayer/hls.js/pull/174

**Comment by johnBartos** [comment]
Blocked by https://github.com/jwplayer/hls.js/pull/180

File: tests/.eslintrc.js [hunk 2]
```diff
@@ -12,9 +13,11 @@ module.exports = {
...
         "describe": false,
-        "expect": false,
-        "it": false,
+        "expect": true,
         "sinon": false,
         "xit": false
     },
+  "rules": {
+      "no-unused-expressions": 0
+  }
```
[the remark this answers is not in the document]
Comment (johnBartos) [reply tests/.eslintrc.js]: It defaults to 4

**Pull request merged**